designer/src/components/condition.tsx

```diff
@@ -423,11 +628,174 @@ const FieldConditionControl = (props: ConditionProps) => {
     }
   };
 
-  const addCondition = () => {
+  const getFieldLabel = (f: FieldType) =>
+    f?.['component-parameters']?.InputLabelProps?.label ||
+    f?.['component-parameters']?.name ||
+    '<unlabeled>';
+
+  /* Checks if a field has predefined options */
+  const isPredefinedOptions = (fieldDef: FieldType | null): boolean => {
+    if (!fieldDef) return false;
+    return ['Select', 'RadioGroup', 'MultiSelect', 'Checkbox'].includes(
+      fieldDef['component-name']
+    );
+  };
+
+  /* Filter the allowed operators based on field type */
+  const getAllowedOperators = (fieldDef: FieldType | null) => {
+    return isPredefinedOptions(fieldDef)
+      ? ['equal', 'not-equal']
```

Comment:
If the target field is a multi-select the value will be an array so we need 'contains'. I note that we don't have 'does not contain' as a comparator so we should add that too. 
so this should be ['equal', 'not-equal'] or ['contains', 'does not contain'].

Reply:
Ah, we do have does-not-contain so it should appear in your longer list as well.